File: financial-planning-main/FinancialPlanningBE/FinancialPlanning.WebAPI/Models/Term/TermViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace FinancialPlanning.WebAPI.Models.Term
{
    public class TermViewModel(Guid id, string termName, Guid creatorId, int duration, DateTime startDate, DateTime planDueDate, DateTime reportDueDate, string status)
    {
       
        public Guid Id { get; set; } = id;

        [Required(ErrorMessage = "Term name is required")]
        public string TermName { get; set; } = termName;

        [Required(ErrorMessage = "Creator ID is required")]
        public Guid CreatorId { get; set; } = creatorId;

        [Required(ErrorMessage = "Duration is required")]
        [RegularExpression("^[1,3,6]$", ErrorMessage = "Duration must be set to 1, 3, or 6")]
        public int Duration { get; set; } = duration;

        [Required(ErrorMessage = "Start date is required")]
        public DateTime StartDate { get; set; } = startDate;

        [Required(ErrorMessage = "Plan due date is required")]
        public DateTime PlanDueDate { get; set; } = planDueDate;

        [Required(ErrorMessage = "Report due date is required")]
        public DateTime ReportDueDate { get; set; } = reportDueDate;

        // [Required(ErrorMessage = "Status is required")]
        // [Range(1, 3, ErrorMessage = "Status must be set to 1, 2, or 3")]
        public string Status { get; } = status;
    }
}

File: financial-planning-main/FinancialPlaningUI/FinancialPlaning/src/app/models/termview.model.ts
export class TermViewModel {
  id: string;
  termName: string;
  creatorId: string;
  duration: number;
  startDate: string;
  planDueDate: string;
  reportDueDate: string;
  status: string;
  user: any; // You might want to define a UserViewModel here if needed

  constructor(data: any = {}) {
    this.id = data.id || '';
    this.termName = data.termName || '';
    this.creatorId = data.creatorId || '';
    this.duration = data.duration || 0;
    this.startDate = data.startDate || '';
    this.planDueDate = data.planDueDate || '';  
    this.reportDueDate = data.reportDueDate || '';
    this.status = data.status || 0;
    this.user = data.user || null;
  }
}
